src/basic/cgroup-util.c

```diff
@@ -140,6 +142,12 @@ int cg_read_pidref(FILE *f, PidRef *ret) {
                         return 0;
                 }
 
+                if (pid == 0)
+                        /* Un unmappable PID. We certainly cannot create a pidref for it,
+                         * so return an error. (Also 0 would be interpreted as us by
+                         * pidref_set_pid(), which we cannot allow to happen.) */
+                        return -ESRCH;
```

Comment:
Returning an error here would not solve the underlying issue the original patch attempted to solve.
It would now just fail with -ESRCH instead of -EIO, since the paths that fail most loudly on WSL all call cg_read_pidref.

Hence the change to skip these pids by default, and only return them when explicitly requested.

The failures there mostly happen in append_cgroup() in dbus-unit.c, but there is multiple functions that are candidates for such failures.

Reply:
Yeah, the original issue was about `GetUnitProcesses` returning error. So `append_cgroup` must ignore `-ESRCH`.

Reply:
Thanks. I changed it to ignore those PIDs. I think this is the right thing to do for all callers, who already sometimes ignore pids for various reasons.